parcels/kernels/advectiondiffusion.py

```diff
@@ -0,0 +1,230 @@
+    ay = fieldset.V[time, particle.depth, particle.lat, particle.lon] + dKdy
+    by = math.sqrt(2 * fieldset.Kh_meridional[time, particle.depth, particle.lat, particle.lon])
+
+    # Particle positions are updated only after evaluating all terms.
+    particle.lon += ax * particle.dt + bx * dWx
+    particle.lat += ay * particle.dt + by * dWy
+
+
+def AdvectionDiffusionMilstein1(particle, fieldset, time):
+    """Kernel for simple advection-diffusion, solved using the Milstein scheme at first order.
+    The Milstein scheme is superior to the Euler-Maruyama scheme, experiencing
+    less spurious background diffusivity by including extra correction
+    terms that are computationally cheap.
+
+    Assumes that fieldset has fields `Kh_zonal` and `Kh_meridional`
+    and variable `fieldset.dres`, setting the resolution for the central difference
+    gradient approximation. This should be at least an order of magnitude
+    less than the typical grid resolution.
+
+    The Wiener increment `dW` should be normally distributed with zero
+    mean and a standard deviation of sqrt(dt). Instead, here a uniform
+    distribution with the same mean and std is used for efficiency and
+    to keep random increments bounded. This substitution is valid for
+    the convergence of particle distributions. If convergence of
+    individual particle paths is required, use normally distributed
+    random increments instead. See Gräwe et al (2012)
+    doi.org/10.1007/s10236-012-0523-y for more information.
+    """
+    # Wiener increment with zero mean and std of sqrt(dt)
+    dWx = random.uniform(-1., 1.) * math.sqrt(math.fabs(particle.dt) * 3)
+    dWy = random.uniform(-1., 1.) * math.sqrt(math.fabs(particle.dt) * 3)
+
+    Kxp1 = fieldset.Kh_zonal[time, particle.depth, particle.lat, particle.lon + fieldset.dres]
+    Kxm1 = fieldset.Kh_zonal[time, particle.depth, particle.lat, particle.lon - fieldset.dres]
+    dKdx = (Kxp1 - Kxm1) / (2 * fieldset.dres)
+
+    bxp1 = math.sqrt(2 * Kxp1)
+    bxm1 = math.sqrt(2 * Kxm1)
+    dbdx = (bxp1 - bxm1) / (2 * fieldset.dres)
+
+    ax = fieldset.U[time, particle.depth, particle.lat, particle.lon] + dKdx
+    bx = math.sqrt(2 * fieldset.Kh_zonal[time, particle.depth, particle.lat, particle.lon])
+
+    Kyp1 = fieldset.Kh_meridional[time, particle.depth, particle.lat + fieldset.dres, particle.lon]
+    Kym1 = fieldset.Kh_meridional[time, particle.depth, particle.lat - fieldset.dres, particle.lon]
+    dKdy = (Kyp1 - Kym1) / (2 * fieldset.dres)
+
+    byp1 = math.sqrt(2 * Kyp1)
+    bym1 = math.sqrt(2 * Kym1)
+    dbdy = (byp1 - bym1) / (2 * fieldset.dres)
+
+    ay = fieldset.V[time, particle.depth, particle.lat, particle.lon] + dKdy
+    by = math.sqrt(2 * fieldset.Kh_meridional[time, particle.depth, particle.lat, particle.lon])
+
+    # Particle positions are updated only after evaluating all terms.
+    particle.lon += ax * particle.dt + bx * dWx + \
+        0.5 * bx * dbdx * (dWx**2 - particle.dt)
```

Comment:
Does `**2` work in JIT? Normally, we use `math.pow(.., 2)` in Kernels.

Reply:
Yes, seems to give no trouble. Particle trajectories are the same in JIT and Scipy for my test runs.